### DISCUSSION TITLE
Code Reusability

### DISCUSSION THREAD

**Discussion by cooper-roper** [climate-cooperative/home-energy-calculator#33] [Q&A]
We have custom components(e.g. CounterItem, CounterQuestion, ImageQuestion, etc.) in this calculator. Over time, we will develop more calculators [(1)](https://github.com/climate-cooperative/hvac-savings-calculator) [(2)](https://github.com/climate-cooperative/heat-pump-water-heater-savings-calculator) that will reuse the same components. The team has ideas as to how we should handle code reusability but what suggestions does the community have?

**Comment by rgraue** [comment]
My initial thought is "will the costs out weight the value?". Kinda vague, but I think an important issue to bring up. A stand alone library introduces:
1. Another code base to maintain.
2. Libraries are much "harder" to create from a user perspective. Think "how do I want other people to use this", "what should it do and more importantly what SHOULDN'T it do"
3. A whole different versioning and publishing strategy
4. Adds complexity that otherwise wouldn't have to be calculated for. "My simple one line change, is now many in multiple repos"

All this not to say that a shared component library isn't a good idea. They're actually quite common in enterprise solutions. However, someone has to play devil's advocate; and today... it's me :)
